contrib/inventory/netbox.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python
+
+# (c) 2017, Ahmed AbouZaid <http://aabouzaid.com/>
+#
+# This file is part of Ansible,
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+import yaml
+import argparse
+import requests
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+# Script.
+def cli_arguments():
+    """Script cli arguments.
+    By default Ansible calls "--list" as argument.
+    """
+
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("-c", "--config-file",
+                        default=os.getenv("NETBOX_CONFIG_FILE", "netbox.yml"),
+                        help="""Path for script's configuration. Also "NETBOX_CONFIG_FILE"
+                                could be used as env var to set conf file path.""")
+    parser.add_argument("--list", help="Print all hosts with vars as Ansible dynamic inventory syntax.",
+                        action="store_true")
+    parser.add_argument("--host", help="Print specific host vars as Ansible dynamic inventory syntax.",
+                        action="store")
+    arguments = parser.parse_args()
+    return arguments
+
+
+# Utils.
+def reduce_path(source_dict, key_path):
+    for key in key_path:
+        if isinstance(source_dict.get(key), dict) and len(key_path) > 1:
+            source_dict = source_dict.get(key)
+            key_path = key_path[1:]
+            reduce_path(source_dict, key_path)
+        else:
+            return source_dict[key]
+
+
+def get_value_by_path(source_dict, key_path, ignore_key_error=False):
+    """Get key value from nested dict by path.
+
+    Args:
+        source_dict: The dict that we look into.
+        key_path: The path of key in dot notion. e.g. "parent_dict.child_dict.key_name"
+        ignore_key_error: Ignore KeyError if the key not found in provided path.
+
+    Returns:
+        If key found in provided path, it will be returned.
+        If not, None will be returned.
+    """
+
+    try:
+        key_output = reduce_path(source_dict, key_path)
+    except KeyError as key_name:
+        if ignore_key_error:
+            key_output = None
+        else:
+            print("The key %s is not found. Please remember, Python is case sensitive." % key_name)
```

Comment:
errors should go to stderr, you can just use sys.exit('message") to do both in one statement